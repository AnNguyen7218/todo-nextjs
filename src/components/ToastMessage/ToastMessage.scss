@import 'config.fonts';
@import 'config.gutters';
@import 'config.colors';

.container {
  display: flex;
}

.icon {
  width: 2rem;
  height: 2rem;
}

.message {
  flex: 1;
  font-family: $fontPrimary;
  @include small;
  margin-left: gutter(8);
  margin-right: gutter(8);
  align-self: center;
}

.color {
  &--success {
    color: $toast-success-text;
  }

  &--error {
    color: $toast-error-text;
  }

  &--warning {
    color: $toast-warning-text;
  }
}
