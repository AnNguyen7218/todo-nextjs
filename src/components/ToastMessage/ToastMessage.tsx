import React from 'react';
import cx from 'classnames';
import CheckCircle from 'assets/icons/ic-check-circle.svg';
import ErrorCircle from 'assets/icons/ic-error-circle.svg';
import Warning from 'assets/icons/ic-warning.svg';
import Close from 'assets/icons/ic-close.svg';
import css from './ToastMessage.scss';

type Props = {
  message: string;
  theme?: ToastTheme;
  closeToast: () => void;
};

const getIconByTheme = (theme: ToastTheme) => {
  switch (theme) {
    case 'success':
      return CheckCircle;
    case 'error':
      return ErrorCircle;
    case 'warning':
      return Warning;
  }
};

const ToastMessage: React.FC<Props> = ({
  message,
  theme = 'success',
  closeToast,
}) => {
  const Icon = getIconByTheme(theme);
  return (
    <div className={css['container']}>
      <Icon className={cx(css['icon'], css[`color--${theme}`])} />
      <p className={cx(css['message'], css[`color--${theme}`])}>{message}</p>
      <Close
        onClick={closeToast}
        className={cx(css['icon'], css[`color--${theme}`])}
        data-testid="close-icon"
      />
    </div>
  );
};

export default ToastMessage;
