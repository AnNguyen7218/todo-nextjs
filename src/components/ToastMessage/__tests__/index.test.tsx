import React from 'react';
import { render, cleanup } from '@testing-library/react';
import ToastMessage from '../ToastMessage';

afterEach(cleanup);

describe('ToastMessage Component', () => {
  test('ToastMessage should match snapshot', () => {
    const { container } = render(
      <>
        <ToastMessage
          message={'Test message'}
          closeToast={() => null}
          theme="success"
        />
        <ToastMessage
          message={'Test message'}
          closeToast={() => null}
          theme="error"
        />
        <ToastMessage
          message={'Test message'}
          closeToast={() => null}
          theme="warning"
        />
      </>
    );
    expect(container).toMatchSnapshot();
  });
});
