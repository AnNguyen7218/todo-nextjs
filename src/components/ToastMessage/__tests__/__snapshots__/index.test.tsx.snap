// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToastMessage Component ToastMessage should match snapshot 1`] = `
<div>
  <div
    class="container"
  >
    <ic-check-circle.svg
      classname="icon color--success"
    />
    <p
      class="message color--success"
    >
      Test message
    </p>
    <ic-close.svg
      classname="icon color--success"
      data-testid="close-icon"
    />
  </div>
  <div
    class="container"
  >
    <ic-error-circle.svg
      classname="icon color--error"
    />
    <p
      class="message color--error"
    >
      Test message
    </p>
    <ic-close.svg
      classname="icon color--error"
      data-testid="close-icon"
    />
  </div>
  <div
    class="container"
  >
    <ic-warning.svg
      classname="icon color--warning"
    />
    <p
      class="message color--warning"
    >
      Test message
    </p>
    <ic-close.svg
      classname="icon color--warning"
      data-testid="close-icon"
    />
  </div>
</div>
`;
